iree/turbine/kernel/wave/decompose_scan_ops.py

```diff
@@ -0,0 +1,130 @@
+import math
+from operator import ge
+from typing import Any, Callable, Optional
+
+import sympy
+import torch.fx as fx
+
+from .._support.dtype import i1
+
+from .._support.tracing import CapturedTrace
+from ..ops.wave_ops import (
+    Add,
+    Cumsum,
+    CustomOp,
+    NewRegister,
+    ScanOp,
+    SelectOp,
+    ShuffleOp,
+    get_custom,
+)
+from ..wave.constraints import HardwareConstraint
+from .utils.graph_utils import DCE
+
+
+def get_graph_node(custom: CustomOp, graph: fx.Graph) -> fx.Node:
+    custom.add_to_graph(graph)
+    return custom.fx_node
+
+
+def get_register_as_graph_node(
```

Comment:
NIT: I think for now may be more compact/straight forward if we use `get_graph_node(NewRegister(shape, dtype, value), graph)` as opposed to having this helper function